compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -459,5 +460,39 @@ object Completion {
     /** Both term and type symbols are allowed */
     val Import: Mode = new Mode(4) | Term | Type
   }
+
+  private val bslash = '\\'
+  private val isDot = (x: Char) => x == '.'
+  private val brackets = List('[',']','(',')','{','}')
+
+  def label(name: Name): String = {
+
+    def maybeQuote(name: Name, recurse: Boolean): String =
+      if (recurse && name.isTermName)
+        name.asTermName.qualToString(maybeQuote(_, true), maybeQuote(_, false))
+      // initially adapted from
+      // https://github.com/scala/scala/blob/decbd53f1bde4600c8ff860f30a79f028a8e431d/
+      //   src/reflect/scala/reflect/internal/Printers.scala#L573-L584
+      else if (name == nme.CONSTRUCTOR) "this"
+      else {
+        val decName = name.decode.toString
+        val hasSpecialChar = decName.exists { ch =>
+          brackets.contains(ch) || ch.isWhitespace || isDot(ch)
```

Comment:
I agree, but maybe this can be fixed later? As soon as this PR is merged, this should allow to merge https://github.com/com-lihaoyi/Ammonite/pull/1150 (using dotty nighlies).